### PR TITLE
JSAAL-66 investments backend

### PR DESCRIPTION
**Pull request by FairyWind**

### CODE REVIEW

File: src/controller/investments.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+const { User } = require('../models');
 
-investments.get('/user/:uid', (req, res) => {
-  const { uid } = req.params;
-  const userData = userInvestments.find((data) => data.userId === uid);
-  if (userData) {
-    return res.send(userData);
+investments.get('/user', async (req, res) => {
+  const { _id } = req.user;
+  const user = await User.findById(_id, 'investments').exec();
```
Comment (ajaxjiang96) [inline src/controller/investments.js]:
```suggestion
  const user = await User.findById(_id, 'investments');
```

**Review by ajaxjiang96 — APPROVED** [review]

File: src/controller/investments.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+  try {
+    const { _id } = req.user;
+    const user = await User.findById(_id, 'investments');
+    if (user) {
+      return res.send(user);
+    }
+    return res.sendStatus(404);
+  } catch (err) {
+    if (err instanceof TypeError) {
+      return res.sendStatus(401);
```
Comment (ikarasz) [inline src/controller/investments.js]: does this mean authentication error?

File: src/controller/investments.js [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
-  if (userData) {
-    return res.send(userData);
+investments.get('/user', async (req, res) => {
+  try {
+    const { _id } = req.user;
+    const user = await User.findById(_id, 'investments');
+    if (user) {
+      return res.send(user);
+    }
+    return res.sendStatus(404);
```
Comment (ikarasz) [inline src/controller/investments.js]: how can we get 404 with a valid token? We generate the token during login, which means there _is_ a user with this id.

File: src/controller/users.js [hunk 4]
```diff
@@ -15,7 +14,7 @@ const loginUser = async (req, res) => {
     const auth = User.authenticate();
     const { user, error } = await auth(username, password);
     if (user) {
-      const accessToken = jwt.sign({ name: user.username, exp: expirationDate }, secret);
+      const accessToken = jwt.sign(user.toJSON(), secret);
```
Comment (ikarasz) [inline src/controller/users.js]: Dont we want to set expiry date for the tokens? It is really dangerous. I would go with an increased expiry date instead

File: src/controller/investments.js [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
 
-investments.get('/user/:uid', (req, res) => {
-  const { uid } = req.params;
-  const userData = userInvestments.find((data) => data.userId === uid);
-  if (userData) {
-    return res.send(userData);
+investments.get('/user', jwtVerifier({ secret }), async (req, res) => {
+  try {
+    const { _id } = req.user;
+    const user = await User.findById(_id, 'investments');
```
Comment (ikarasz) [inline src/controller/investments.js]: are you sure we don't need to transfer the data? Don't we have an `_id` in the user document? All the keys are in `camelCase` and has only the mandatory fields? Aren't there any mongo related fields in the document?

File: src/controller/investments.js [hunk 6]
```diff
@@ -1,15 +1,33 @@
...
-investments.get('/user/:uid', (req, res) => {
-  const { uid } = req.params;
-  const userData = userInvestments.find((data) => data.userId === uid);
-  if (userData) {
-    return res.send(userData);
+investments.get('/user', jwtVerifier({ secret }), async (req, res) => {
+  try {
+    const { id } = req.user;
+    const user = await User.findOne({ _id: id }, '-_id investments');
+    const result = user.investments.map((stock) => {
```
Comment (ikarasz) [inline src/controller/investments.js]: how about something like this?
```js
user.investments.map(({ _id, ...stock }) => ({ id: _id, ...stock }))
```

**Review by ikarasz — CHANGES_REQUESTED** [review]

**Review by ikarasz — APPROVED** [review]

**Pull request merged**